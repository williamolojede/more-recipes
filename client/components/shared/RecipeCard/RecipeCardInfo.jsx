import React from 'react';

import { recipeCardInfoPropTypes } from '../../../config/proptypes';
import defaultConfig from '../../../config/default';

import RecipeStats from '../RecipeStats.jsx';


const RecipeCardInfo = (props) => {
  const { User, upVoteCount, downVoteCount, favoriteCount, viewCount } = props.info;

  return (
    <div className="card-action recipe-card__info">
      {/* if user img is present show it or the default if not */}
      <img
        src={User.imgUrl || defaultConfig.userImgUrl}
        alt={User.fullname}
        title={User.fullname}
        className="recipe-card__info---img"
      />
      <div className="recipe-card__info">
        <span><span className="recipe-card__info-by">Recipe by</span> {User.fullname}</span>
        <RecipeStats />
        <ul className="recipe-card__info--stats">
          <li className="stats__favorite">
            <i className="mdi mdi-heart" />
            <span>{favoriteCount}</span>
          </li>
          <li className="stats__vote-up">
            <i className="mdi mdi-thumb-up" />
            <span>{upVoteCount}</span>
          </li>
          <li className="stats__vote-down">
            <i className="mdi mdi-thumb-down" />
            <span>{downVoteCount}</span>
          </li>
          <li className="stats__view">
            <i className="mdi mdi-eye" />
            <span>{viewCount}</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

RecipeCardInfo.propTypes = recipeCardInfoPropTypes;
export default RecipeCardInfo;
