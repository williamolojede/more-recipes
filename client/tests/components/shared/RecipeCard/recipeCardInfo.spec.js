import React from 'react';
import RecipeCardInfo from '../../../../components/shared/RecipeCard/RecipeCardInfo.jsx';
import recipesMock from '../../../__mocks__/recipes.mock';

function setup() {
  const {
    User,
    upVoteCount,
    downVoteCount,
    favoriteCount,
    viewCount
  } = recipesMock.recipes[5];

  const props = {
    info: {
      User,
      upVoteCount,
      downVoteCount,
      favoriteCount,
      viewCount
    }
  };
  const wrapper = shallow(<RecipeCardInfo {...props} />);

  return {
    wrapper
  };
}

describe('Recipe Card Info', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
