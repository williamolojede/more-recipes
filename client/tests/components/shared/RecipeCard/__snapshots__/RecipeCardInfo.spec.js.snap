// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card Info should render with right amount of elements 1`] = `
<div
  className="card-action recipe-card__info"
>
  <UserImg
    type="inRecipeCard"
    user={
      Object {
        "fullname": "Email Mail",
        "id": 11,
        "username": null,
      }
    }
  />
  <div
    className="recipe-card__info"
  >
    <RecipeStats
      stats={
        Object {
          "downVoteCount": 0,
          "favoriteCount": 0,
          "upVoteCount": 0,
          "viewCount": 1,
        }
      }
    />
  </div>
</div>
`;
